### DISCUSSION TITLE
Github API-compare-two-commits: pagination seems to don't work

### DISCUSSION THREAD

**Discussion by etsybaev** [community/community#33505] [Apps, API and Webhooks]
Hi.

Not sure if that is the right place, but I didn't find the option to report a bug.

I.m currently trying to fix one issue in our project PR's action.
We need to find changed files in PRs to do some processing in action and add an automatic comment in PR.. It works fine when you create a PR from the branch in the same repo, but have some issues when PR comes from a forked repository. In this case git diff says that there is no breach.

I found this API https://docs.github.com/en/rest/commits/commits#compare-two-commits that could help to fix the issue and should allow such comparison
 E.x. 
GET https://api.github.com/repos/airbytehq/airbyte/compare/master...etsybaev:etsybaev-test-fork-branch?per_page=1&page=2

The PR contains 3 files changed, so I would expect to have 3 pages when use per_page=1, but all files still would be shown in the first page and the second page would not contain any "files" section.. This is just a simplified example. It gets worst if PR contains hundreds of files (for example after some refactioring). In this case there would be a limited amount of files in the list (+- 200 files) and others would be missed at all. 

 I.e. "per_page" and "page" query params are completely ignored.

Would be thankful for any feedback on my observation from the GitHub team.
Many thanks in advance!

**Comment by ryanrmcintyre** [comment]
Having the same issue, was this ever resolved?
